File: Listas Dikastis/Lista 1/A-deteccao_de_replica.py
def isReplica(palavra, num, letra):
    return (palavra == num*letra)



print(isReplica("ee", 2, "e"))
print(isReplica("uruu", 3, "u"))
print(isReplica("xxx", 3, "y"))

'''
Case: 1
Input
ee
2
e
Output
True

Case: 2
Input
uruu
3
u
Output
False

Case: 3
Input
xxx
3
y
Output
False '''

File: Listas Dikastis/Lista 1/A-deteccao_de_replica.hs
isReplica :: String -> Int -> Char -> Bool
isReplica palavra num letra = (palavra == res)
    where
        res = concatena letra num

concatena :: Char -> Int -> String
concatena _ 0 = ""
concatena letra num = [letra] ++ concatena letra (num-1)

main = do
    a <- getLine
    b <- getLine
    c <- getChar
    let result = isReplica a (read b) c
    print result